core/metadata.js

```diff
@@ -675,3 +678,21 @@ function shouldIgnore(
     isFolder: doc.docType === 'folder'
   })
 }
+
+function needsRevMigration(existing /*: Metadata */) /*: boolean %checks */ {
+  return existing.sides == null || existing.sides.rev == null
+}
+
+async function migrateRev(
+  existing /*: Metadata */,
+  { pouch } /*: { pouch: Pouch } */
+) /*: Promise<Metadata> */ {
+  log.info({ path: existing.path, _rev: existing._rev }, 'Migrating _rev')
+  const doc = _.defaultsDeep(
+    { sides: { rev: extractRevNumber(existing) } },
+    existing
+  )
+  const { rev: _rev } = await pouch.put(doc)
+  doc._rev = _rev
+  return doc
```

Comment:
Both mutating & returning new value?

Reply:
Yes I'm not super happy about it but I wanted to avoid another request to Pouch to get the new `doc`.
I'm open to suggestions.

Reply:
This is done via a migration now.